client/src/commands/exportWalletCommand.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+'use strict';
+import { VSCodeBlockchainOutputAdapter } from '../logging/VSCodeBlockchainOutputAdapter';
+import { LogType } from '../logging/OutputAdapter';
+import { UserInputUtil, IBlockchainQuickPickItem } from './UserInputUtil';
+import { FabricWalletRegistryEntry } from '../fabric/FabricWalletRegistryEntry';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import { Reporter } from '../util/Reporter';
+import { WalletTreeItem } from '../explorer/wallets/WalletTreeItem';
+
+export async function exportWallet(walletTreeItem?: WalletTreeItem): Promise<void> {
+
+    const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
+    outputAdapter.log(LogType.INFO, undefined, 'exportWalletCommand');
+
+    let exportedWalletRegistryEntry: FabricWalletRegistryEntry;
+
+    if (!walletTreeItem) {
+        // called from command palette - ask for wallet to export
+        const chosenWallet: IBlockchainQuickPickItem<FabricWalletRegistryEntry> = await UserInputUtil.showWalletsQuickPickBox('Choose a wallet to export', true);
+        if (!chosenWallet) {
+            // User cancelled dialog box
+            return;
+        }
+        exportedWalletRegistryEntry = chosenWallet.data;
+    } else {
+        // Called from the tree
+        exportedWalletRegistryEntry = walletTreeItem.registryEntry;
+    }
+
+    // Ask the user where they want to export it to
+    // set the default path to be the first open workspace folder
+    let defaultPath: string;
+    const workspaceFolders: Array<vscode.WorkspaceFolder> = await UserInputUtil.getWorkspaceFolders();
+    if (workspaceFolders.length > 0) {
+        defaultPath = path.join(workspaceFolders[0].uri.fsPath, exportedWalletRegistryEntry.name);
+    } else {
+        defaultPath = path.join(os.homedir(), exportedWalletRegistryEntry.name);
+    }
+
+    const walletUri: vscode.Uri = await vscode.window.showSaveDialog({
```

Comment:
Worth checking with @EdMoffatt if we want to ask the user if they want to 'Browse' / 'Save' before opening this dialog (particularly from the command)

Reply:
Spoke to Ed and he's happy with it this way, given it's fairly obvious the user has chosen to export